### ISSUE TITLE
Excessive WAN Failovers.

### ISSUE THREAD

**Issue by routersecskisthedude** [Ranger802004/asusmerlin#56]
I am experiencing a very high number of failovers. Logs attached. Please advise.
[wan_failover_logs.txt](https://github.com/user-attachments/files/18132873/wan_failover_logs.txt)

**Comment by Ranger802004** [comment]
This appears to be normal behavior from packet loss / failed pings.  Have you verified your Primary WAN is not failing or the IP you are using is a reliable source.

**Comment by routersecskisthedude** [comment]
It seems to always switch from wan 0 to wan 1 and almost right away switch back to wan 0. I think the switching causes some disruption to the internet at those times not the service. I have tried pinging from the command line on my computer at the time of these issues and the ping times seem very good. I have the latest version Netgear cable modem (with up to date firmware) with consistent 1gbps down 400 Mbps up.

**Comment by Ranger802004** [comment]
1. Try a different target IP for WAN0.
2. Try and increase the ping count.

**Comment by Ranger802004** [comment]
Following up to see if you have any updates on this issue?

**Comment by routersecskisthedude** [comment]
I have the ping count to 7 from 5 and changed the target IP to Google. I still have 3-5 incidents happening per day. Happens with the same characterstics - wan0 is identified as disconnected and within about 30 seconds is fine again. I receive clusters of 4 emails per incident.

**Comment by routersecskisthedude** [comment]
To me that is overall okay, better than 15-20 incidents per day that have happened in the past. Ideally, I would be at 0 for most days.

**Comment by Ranger802004** [comment]
> I changed the ping count to 7 from 5 and changed the target IP to Google. I still have 3-5 incidents happening per day. Happens with the same characterstics - wan0 is identified as disconnected and within about 30 seconds is fine again. I receive clusters of 4 emails per incident.

For some reason your router is losing connectivity to the ping IP during these events.  It could be anything from a physical issue or another script/tool executing a command that is resetting things on the router like firewall / ip rules.  I would recommend to start investigating those possibilities.

**Comment by routersecskisthedude** [comment]
I changed a lot of settings on the router in order to force all DNS requests to go through pihole. Other than that, I am not  sure what else it could be.

**Comment by Ranger802004** [comment]
Do you have any other addons installed?

**Comment by routersecskisthedude** [comment]
I need until the end of the week to check.

**Comment by Ranger802004** [comment]
Following back with you.

**Comment by routersecskisthedude** [comment]
I have email settings, firmware update notification, and shell history installed. Not sure if it is important, but the swap drive is connected by USB 3.0.

**Comment by Ranger802004** [comment]
Those don't sound like they would cause issues but some Entware binaries possibly could.

**Comment by routersecskisthedude** [comment]
I don't believe I have any entware binaries, only from amtm.

**Comment by Ranger802004** [comment]
Have you tried changing the physical cables used for your WAN connections between the CPE devices and the router?

**Comment by routersecskisthedude** [comment]
I changed the cables two days ago, but still had many switches back and forth. I decided to uninstall the tool and disable dual WAN this morning after having WiFi connection issues. I still had internet connectivity issues, however. I think there maybe an issue with my service as you previously suggested. 

I reinstalled this evening. Is there anyway to set it so that when failover happens it stays on WAN1 for 4-5 minutes before switching back. The switch from WAN 0 to WAN 1 to WAN 0 happens over 20-30 seconds which I don't think is good.

**Comment by Ranger802004** [comment]
Not at this time although something I can consider is adding a failback delay timer to the configuration.

**Comment by routersecskisthedude** [comment]
I think it maybe better to use load balancing instead of wan failover.

**Comment by Ranger802004** [comment]
It will certainly operate faster with the script but it also has its limitations as well.  Have you tried contacting your provider to see what is going on with your connection?

**Comment by Ranger802004** [comment]
I added a Failback Delay Timer to the latest beta release (v2.1.3-beta1).  Please test and provide feedback.

**Comment by Ranger802004** [comment]
Following up with you to see if you have any feedback.

**Comment by routersecskisthedude** [comment]
I have not installed it yet. I plan to Sunday when I return. How can I install beta software? I usually use amtm.

**Comment by Ranger802004** [comment]
Enable Dev Mode in the configuration and you it will offer beta updates to you when checking for updates in amtm and the built-in UI method.

**Comment by routersecskisthedude** [comment]
I have it up and running. Set failback delay to 300 seconds. Will see how it works. Hopefully, it also takes Wan0 health into account before failback. On Asus GUI, it has a 'failback trigger condition' to ensure healthy Wan0 before failback. I had set it to '99' successful DNS queries.

**Comment by routersecskisthedude** [comment]
The status reads 'Failover disabled' and it seems no pings are being sent.

**Comment by Ranger802004** [comment]
The logs should be telling you why it in disabled mode.

**Comment by Ranger802004** [comment]
> Status goes from saying `Failover disabled` to `Not Running` after restarting the script.
> 
> Only line in the logs is:
> 
> `Feb 25 09:19:01 wan-failover: Restart - Successfully Restarted wan-failover Process ID(s): 30386 30387 30646`

Do you have any of the Factory WAN failover settings enabled? Is your router configured in Dual WAN mode?

**Comment by Ranger802004** [comment]
> I have it up and running. Set failback delay to 300 seconds. Will see how it works. Hopefully, it also takes Wan0 health into account before failback. On Asus GUI, it has a 'failback trigger condition' to ensure healthy Wan0 before failback. I had set it to '99' successful DNS queries.

If any of the factory failover settings are enabled in the GUI, WAN Failover script stays disabled.

**Comment by routersecskisthedude** [comment]
Here are my settings: 

<img width="766" alt="Image" src="https://github.com/user-attachments/assets/bb12f6d9-71bd-4178-b958-5f2d691b3a37" />

**Comment by routersecskisthedude** [comment]
I unchecked DNS Query and it is working now. I will monitor for a few days and report back

**Comment by Ranger802004** [comment]
So does everything work properly now?

**Comment by routersecskisthedude** [comment]
Surprisingly, I have not had any switches in the past few days. I set the failback to 360 seconds and left all other parameters with their defaults. The only notification I have received is for the daily resets I automate for my secondary WAN which does not affect switching.

**Comment by routersecskisthedude** [comment]
Do you want me to try any artificial tests? Like unplugging the main WAN interface?

**Comment by Ranger802004** [comment]
No that is ok, thank you.  Everything still looking good on your end?

**Comment by routersecskisthedude** [comment]
Yes, same. Everything is very stable. I used to get 3-4 switches on a good day. Now, nothing with the update. No noticeable problems with internet service that would require a switch either.

**Comment by routersecskisthedude** [comment]
Thank you for your help addressing my issue!
